### PR TITLE
Support @forward and @use

### PR DESCRIPTION
**Pull request by TheMagician23**
This PR adds basic support for the @forward and @use rules, which were previously unsupported.

I followed the syntax highlighting rules on the https://sass-lang.com website; the "as" and "with" keywords are highlighted in the same color as "@forward" and "@use", and namespaces are highlighted in the same color as variables.

The only thing that doesn't currently work is parenthesized blocks of SCSS following a "with". I couldn't figure out a way to say "apply the regular syntax rules of the filetype within these parentheses", which feels like the kind of thing that should be easy but probably isn't.

I also updated the example.scss file with usage examples of the newly-supported rules.

This is my first time working with Vim syntax definitions; I was putting things together largely by scanning the built-in help and experimenting, so it's possible I've made a few beginner errors. If so please correct me.

### CODE REVIEW

File: syntax/scss.vim [hunk 1]
```diff
@@ -143,9 +143,19 @@ syn match scssReturn "@return" contained
 syn match scssExtend "@extend" nextgroup=scssExtendedSelector skipwhite containedin=cssMediaBlock
 syn match scssExtendedSelector "[^;]\+" contained contains=cssTagName,cssPseudoClass,scssOptional,scssSelectorChar skipwhite
 syn match scssOptional "!optional" contained
-syn match scssImport "@import" nextgroup=scssImportList
+
+syn match scssImport "@import" nextgroup=scssUrlList
```
Comment (cakebaker) [inline syntax/scss.vim:147]: What's the reason for this change? Did you forget to add the definition for `scssUrlList`? This is the only line it appears on.

Comment (TheMagician23) [reply syntax/scss.vim]: Sorry, that was an artefact left over from an earlier version of the implementation where there was a shared group used by @import, @forward and @use, before I realized I'd have to implement "as" and "with" for @forward and @use. It should be "scssImportList". I've just pushed a commit which fixes this.

**Comment by bdice** [comment]
@TheMagician23 @cakebaker Hi, I just ran across a need for this feature as well. Would it be possible to complete this PR? It looks like the last round of changes was resolved and it's waiting for a review.

**Comment by cakebaker** [comment]
@bdice: Thanks for the reminder, I completely forgot about this PR :|

@TheMagician23: Thanks for your PR and I'm sorry about the delay.

**Comment by bdice** [comment]
Thank you very much @cakebaker! 🚀
